tests/e2e/specs/layers.spec.js

```diff
@@ -300,6 +310,70 @@ describe('Test of layer handling', () => {
                 expect(visibleIds).to.contain(testLayerId)
             })
         })
+        it('keeps the layer settings when changing language', () => {
+            const langBefore = 'en'
+            const langAfter = 'de'
+            let activeLayersConfigBefore
+
+            // Change the layer configurations to non-default values.
+            cy.readWindowValue('store').then((store) => {
+                visibleLayerIds.forEach((layerId) => {
+                    store.dispatch('toggleLayerVisibility', { layerId })
+                    store.dispatch('setLayerOpacity', {
+                        layerId,
+                        opacity: 0.1,
+                    })
+                })
+                cy.fixture('layers.fixture.json').then((fixtures) => {
+                    const layerId = 'test.timeenabled.wmts.layer'
+                    const layerMetadata = fixtures[layerId]
+                    const layerTimestamp = getRandomTimestampFromSeries(layerMetadata)
+                    store.dispatch('setTimedLayerCurrentTimestamp', {
+                        layerId,
+                        timestamp: layerTimestamp,
+                    })
+                })
+            })
```

Comment:
In order to change the app at startup, I would prefer that we leverage the fact that the URL represents the app state too.
So instead of changing store values with dispatch (that will be hard to track if they are renamed later on), maybe use a `cy.goToMapView('lang', { layers: '...' })` such as [here](https://github.com/geoadmin/web-mapviewer/blob/26c89c3b09b863ba8e91054790d3def36d12b3b2/tests/e2e/specs/layers.spec.js#L47-L49)

You can find more information on how to describe all parts of a layer configuration in [this ADR](https://github.com/geoadmin/web-mapviewer/blob/develop/adr/2021_03_16_url_param_structure.md#decision)

Reply:
Good idea! I changed the configuration to only use the goToMapView command.

To avoid running it twice (and also to separate layer settings from language settings) I created a new context ("Language settings in menu") where I put the test.